daemon/containerd/image_list.go

```diff
@@ -241,13 +247,23 @@ func (i *ImageService) singlePlatformImage(ctx context.Context, contentStore con
 		}
 	}
 
+	cfgDesc, err := image.Image.Config(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+	var cfg configLabels
+	if err := readConfig(ctx, contentStore, cfgDesc, &cfg); err != nil {
```

Comment:
LOL, very unrelated, but I was curious if we make use of the `Data` field anywhere in our code; https://github.com/opencontainers/image-spec/blob/v1.1.0-rc4/specs-go/v1/descriptor.go#L38-L41

```go
	// Data is an embedding of the targeted content. This is encoded as a base64
	// string when marshalled to JSON (automatically, by encoding/json). If
	// present, Data can be used directly to avoid fetching the targeted content.
	Data []byte `json:"data,omitempty"`
```

Reply:
I haven't seen any code using that yet TBH. Let me check, if it can be used that would be great actually

Reply:
It doesn't seem to be set by containerd, we might want to take a note of this and see if it can be used really

Reply:
Yup! I could see something along the lines of

```go
if desc.Data != nil {
  // fast path
}
```

That said; I wonder if the Spec is clear enough on verifying `Data` matches the referenced data (would be fun if there's no such check and `Data` would return something very different from the referenced file 😂)

Reply:
Related https://github.com/containerd/containerd/pull/8076

Reply:
Nice, so yes, looks like we can start looking into using that